user/views.py

```diff
@@ -1,17 +1,55 @@
 from django.shortcuts import render, redirect
+from django.contrib.auth.hashers import make_password
+from django.contrib.auth import authenticate, login, logout
+from django.contrib import messages
+from .models import User
 
-# Create your views here.
 
 
 
+def home(request):
+    if request.user.is_authenticated:
+        return render(request,'tweet/home.html')
+    return render(request,'tweet/login.html')
 
-def check_login(request):
+def signup(request):
     if request.user.is_authenticated:
         return redirect('home')
-    return render(request,'tweet/index.html')
+    if request.method == 'GET':
+        return render(request,'tweet/register.html')
+    email = request.POST.get('email')
+    first_name = request.POST.get('first_name')
+    last_name = request.POST.get('last_name')
+    password = make_password(request.POST.get('password'))
+    try:
+        User.objects.create(username=email, password=password, first_name=first_name, last_name=last_name, email=email)
+        messages.success(request, "Login to continue")
+    except Exception as e:
+        messages.error(request, e)
+    return redirect('home')
 
+def signin(request):
+    if request.user.is_authenticated:
+        return redirect('home')
+    if request.method == 'GET':
+        return render(request,'tweet/login.html')
 
-def home(request):
-    if request.user.is_authenticated:        
-        return render(request,'tweet/home.html')
-    return redirect('index')
\ No newline at end of file
+    email = request.POST.get('email')
+    password = request.POST.get('password')
+    select_user = User.objects.filter(username=email)
```

Comment:
why username. Why not email ?

Reply:
we can do with both

Reply:
It feels weird that your are comparing email with username
We should keep it consistent

Reply:
This is fixed now